tools/missing-test-detector/reader_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestReadTestFile(t *testing.T) {
```

Comment:
It would be great to have more tests here, operating on synthetic data, that exercise a variety of cases you expect to find.

Reply:
Done.